azkaban-common/src/main/java/azkaban/AzkabanCommonModule.java

```diff
@@ -78,6 +66,9 @@ protected void configure() {
   public Class<? extends Storage> resolveStorageClassType() {
     final StorageImplementationType type = StorageImplementationType
         .from(this.config.getStorageImplementation());
+    if (type == StorageImplementationType.HDFS) {
+      install(new HadoopModule(this.props));
```

Comment:
why should we install here? this seems irrelevant to this method's intention.

Reply:
I was considering this question where we should put installModule, but can not find a good approach. First off, install module cannot be put under AzkabanServer class when injector is initialized, because Guice.createInjector can not optionally install a module. The second alternative location is inside `configure` method in AzkabanCommonModule. It will incur duplicate code to resolve storage type, which is same logic inside the method `resolveStorageClassType`. The final alternative is just here. Only when we provide HDFSStorage class, we install this module. I have some idea how we can refactor HDFSStorage bindings, but it will incur a big code change, and we can do it in future.